components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts

```diff
@@ -1,26 +1,44 @@
-import { throwError as observableThrowError,  Observable } from 'rxjs';
-
-import { catchError } from 'rxjs/operators';
-import { Injectable } from '@angular/core';
 import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { throwError as observableThrowError, Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
 import { get } from 'lodash/fp';
-import { ChefSessionService } from '../chef-session/chef-session.service';
+
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { ChefSessionService } from 'app/services/chef-session/chef-session.service';
+import * as selectors from 'app/services/projects-filter/projects-filter.selectors';
+import { ProjectsFilterOption } from '../projects-filter/projects-filter.reducer';
 
 @Injectable()
 export class HttpClientAuthInterceptor implements HttpInterceptor {
 
-  constructor(private chefSession: ChefSessionService) {}
+  private projects: string;
+
+  constructor(
+    private chefSession: ChefSessionService,
+    private store: Store<NgrxStateAtom>
+  ) {
+    this.store.select(selectors.options)
+      .subscribe((options: ProjectsFilterOption[]) => {
+        this.projects = options.filter(p => p.checked).map(p => p.value).join(', ');
+      });
+  }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    let headers = request.headers.set('Authorization', `Bearer ${this.chefSession.id_token}`);
+    if (this.projects) {
+      headers = headers.set('projects', this.projects);
```

Comment:
one idea @scottopherson gave me: 
```
if (thisEndpointDoesntNeedOrSupportFilteringByProjects) {
      return next.handle(req.clone());
    }
```

we could write the helper `thisEndpointDoesntNeedOrSupportFilteringByProjects` and keep track of all the APIs that shouldn't be filtered inside it

Reply:
Didn't see this PR until this morning. I'd also recommend having separate `Auth` and `ProjectsFilter`  interceptors in the request chain, similar to the branch I was playing with last night:

https://github.com/chef/automate/compare/sc/projects-filter-data#diff-673f4abc0b622315151c256ba3238a8b
https://github.com/chef/automate/compare/sc/projects-filter-data#diff-b99ce0ca235bf0716d691eab97a28bf5R311

Reply:
```
@Injectable()
export class ProjectsFilterInterceptor implements HttpInterceptor {
  constructor(private projectsFilter: ProjectsFilterService) {}

  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {

    if (thisEndpointDoesntNeedOrSupportFilteringByProjects) {
      // pass the request through
      return next.handle(req.clone());
    }

    return this.projectsFilter.selectionValue$.pipe(
      first(),
      flatMap(selectionValue => {
        const headers = request.headers.set('projects', selectionValue);

        if (selectionValue === ProjectsConstants.ALL_RESOURCES_VALUE) {
          headers.delete('projects');
        }

        return next.handle(request.clone({ headers }));
      })
    );
  }
}
```